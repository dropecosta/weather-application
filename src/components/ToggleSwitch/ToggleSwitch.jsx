import React, { useState } from 'react';
import './ToggleSwitch.scss';

const ToggleSwitch = (props) => {
  const [toggled, setToggled] = useState(true);
  return (
    <label
      className="toggle"
      onClick={() => {
        setToggled((checked) => !checked);
        props.onClick();
      }}
    >
      {toggled && <span className="on">C</span>}
      {!toggled && <span className="off">F</span>}
      <div
        style={{
          transform: toggled ? "translateX(28px)" : "translateX(0px)",
        }}
      ></div>
    </label>
  );
};

export default ToggleSwitch;
