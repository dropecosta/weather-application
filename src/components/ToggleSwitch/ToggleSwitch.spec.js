import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import ToggleSwitch from "./ToggleSwitch";

describe("ToggleSwitch component tests", () => {
  it("should render APP component with container class", () => {
    const { container } = render(<ToggleSwitch />);
    expect(container.firstChild).toHaveClass("toggle");
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<ToggleSwitch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
