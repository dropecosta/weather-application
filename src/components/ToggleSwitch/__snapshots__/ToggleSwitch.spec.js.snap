// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch component tests should match the snapshot 1`] = `
<label
  className="toggle"
  onClick={[Function]}
>
  <span
    className="on"
  >
    C
  </span>
  <div
    style={
      Object {
        "transform": "translateX(28px)",
      }
    }
  />
</label>
`;
